pyomo/core/base/block.py

```diff
@@ -1391,15 +1391,17 @@ def _component_data_iter(self, ctype=None, active=None, sort=False):
             else:
                 _items = tuple()
 
+            sorter = lambda x: x
             if _sort_indices:
-                _items = sorted(_items, key=itemgetter(0))
+                sorter = lambda x: sorted(x, key=itemgetter(0))
+
             if active is None or not isinstance(comp, ActiveIndexedComponent):
-                for idx, compData in _items:
-                    yield (name, idx), compData
+                filtered = _items
             else:
-                for idx, compData in _items:
-                    if compData.active == active:
-                        yield (name, idx), compData
+                filtered = filter(lambda tup: tup[1].active == active, _items)
+
+            for idx, compData in sorter(filtered):
+                yield (name, idx), compData
```

Comment:
What do you think of the following?

```
if active is not None and isinstance(comp, ActiveIndexedComponent):
    _items = filter(lambda tup: tup[1].active == active, _items)

if _sort_indices:
    _items = sorted(_items, key=itemgetter(0))

for idx, compData in _items:
    yield (name, idx), compData
```

I thought perhaps it's a bit simpler as we don't need to deal with as many lambdas and long lived variables?

Reply:
Much nicer, thanks!